Lib/typing.py

```diff
@@ -734,10 +734,20 @@ class ForwardRef(_Final, _root=True):
     def __init__(self, arg, is_argument=True, module=None, *, is_class=False):
         if not isinstance(arg, str):
             raise TypeError(f"Forward reference must be a string -- got {arg!r}")
+
+        # If we do `def f(*args: *Ts)`, then we'll have `arg = '*Ts'`.
+        # Unfortunately, this isn't a valid expression on its own, so we
+        # do the unpacking manually.
+        if arg[0] == '*':
+            arg_to_compile = f'next(iter({arg[1:]}))'
```

Comment:
What if next or iter is shadowed in the current scope? We can probably do something like `(*arg,)[0]` instead, which doesn't rely on any names.

Reply:
Good catch! Fixed.